src/engine/shape-builder.js

```diff
@@ -338,11 +394,34 @@ export default class ShapeBuilder extends DrawEngine {
       if (selectedShape && !selectedShape.selected) {
         const selectEventObj = selectShape(selectedShape, shapes, this._selectedShapes, this._selectStyle, this._xformStyle, selectedInfo)
         this.fire(EventConstants.SELECTION_CHANGED, selectEventObj)
+      } else {
+        // If user clicks anywhere outside then allow the movement of Base Map (Parents Container)
+        this._makeParentElementMovable()
+      }
+    }
+    // Added Support for Double click
+    if (event.touches) {
+      if (event.cancelable) {
+        event.preventDefault()
+      }
+      if ((Date.now() - this.firstTapTime) < DOUBLE_CLICK_DELAY) {
+        this._dblclickCB(event)
```

Comment:
This looks ok. Did you verify that this works like a mouse double-click event. In particular I'm wondering if a mouse double-click event actually runs the mousedown/mouseup twice first before firing the double click event. If that's true, then this looks right because the touchstart/touchend would be fired twice before `_dblclickCB` is called.